app/models/event_stream.rb

```diff
@@ -82,19 +92,34 @@ def self.group_name(group)
     group.nil? ? 'Other' : group[:name]
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group
     return @group unless @group.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_level
     return @group_level unless @group_level.nil?
     @group, @group_level = self.class.group_and_level(event_type)
     @group_level
   end
 
+  # TODO: Consider moving since this is EmsEvent specific. group, group_level and group_name exposed as a virtual columns for reports/api.
   def group_name
     @group_name ||= self.class.group_name(group)
   end
+
+  def self.group_and_level_details
+    {'description' => description}.merge!(group_names_and_levels)
+  end
+
+  private_class_method def self.group_names_and_levels
+    raise NotImplementedError, _("group_names_and_levels must be implemented in a subclass")
+  end
+
+  def self.event_stream_options
```

Comment:
wonder if this should be named `timeline_options`, so it is clear what the purpose is for?

Reply:
Good point.  I guess that's the UI usage of it.  I suppose various events could be presented in any format, not just a timeline.  For example, you could just tableize the events, level, group/category, etc.

Reply:
Good idea.  I can't come up with a better name so I'm fine with using the only client of this.